src/core/misc.c

```diff
@@ -261,6 +261,26 @@ char *gslist_to_string(GSList *list, const char *delimiter)
 	return ret;
 }
 
+gboolean gslist_is_unique(GSList *list)
+{
+	const char *prev;
+
+	if (list == NULL || list->next == NULL)
+		return TRUE;
+
+	prev = (const char *)list->data;
+	list = list->next;
+
+	while (list != NULL) {
+		if (strcmp(list->data, prev) != 0)
+			return FALSE;
+		prev = (const char *)list->data;
```

Comment:
Since this line can only be called if `list->data` is equal to `prev`, then this operation will simply set `prev` to the same to a pointer to a new string, which contains the same data.